components/common/src/command/package/install.rs

```diff
@@ -220,15 +314,19 @@ struct InstallTask<'a> {
 
 impl<'a> InstallTask<'a> {
     fn new(
+        install_mode: &'a InstallMode,
         url: &str,
+        channel: Channel<'a>,
         product: &str,
         version: &str,
         fs_root_path: &'a Path,
         artifact_cache_path: &'a Path,
         key_cache_path: &'a Path,
     ) -> Result<Self> {
         Ok(InstallTask {
+            install_mode: install_mode,
             depot_client: Client::new(url, product, version, Some(fs_root_path))?,
+            channel: channel,
```

Comment:
Can this (and the lines around it) make use of [field init shorthand](https://doc.rust-lang.org/book/second-edition/ch05-01-defining-structs.html#using-the-field-init-shorthand-when-variables-and-fields-have-the-same-name)?

Reply:
It's definitely clever, but imo, field init shorthand makes the code harder to read.  Just a personal preference, though.

Reply:
If that's the team consensus, that would be fine. As with most things style, I think the lion's share of the benefit comes from consistency which reduces cognitive load.

Reply:
Yeah, I could have but I tend to preserve the same style of the surrounding code where possible. In this case I did consider it, but I didn't want to change the entire function call to make it more modern (and consistent) and didn't want to have 2 styles going on (older and modern but inconsistent).

If this was all new code however, I'd likely do just that!